osu.Desktop/DiscordRichPresence.cs

```diff
@@ -52,8 +66,12 @@ private void load(OsuConfigManager config)
             };
 
             client.OnReady += onReady;
+            client.OnError += (_, e) => Logger.Log($"An error occurred with Discord RPC Client: {e.Code} {e.Message}", LoggingTarget.Network, LogLevel.Error);
 
-            client.OnError += (_, e) => Logger.Log($"An error occurred with Discord RPC Client: {e.Code} {e.Message}", LoggingTarget.Network);
+            // A URI scheme is required to support game invitations, as well as informing Discord of the game executable path to support launching the game when a user clicks on join/spectate.
+            client.RegisterUriScheme();
```

Comment:
[We can provide a filename ourselves](https://github.com/Lachee/discord-rpc-csharp/blob/433897b5f09593124ab38718b0e9e58d33457680/DiscordRPC/DiscordRpcClient.cs#L868) if that's the worry, but I feel like that would just complicate things.

This seems to update the registry/`.desktop`/whatever **every time** as opposed to just initializing the values **once**, so I imagine what will happen is when osu! updates it'll just smoothly update everything to the new executable when this gets called on startup.

I trust this to work.

Reply:
I have tested that this retrieves the file location correctly using the `Process.GetCurrentProcess().MainModule.FileName` path. 

That being said, it points to the executable inside the release folder rather than the shared one, so there's a very rare scenario wherein the game is updated, the user exits, and then tries to launch the game using Discord (making Discord launch the old version instead of the new version).

The fact that the osu! executable location changes every version hasd already been causing other issues such as https://github.com/ppy/osu/issues/10588, https://github.com/ppy/osu/issues/10550, etc., therefore I'll consider this a non-issue for the time being.